app.py

```diff
@@ -230,6 +251,39 @@ def delete_user():
     return redirect("/signup")
 
 
+@app.route('/users/add_like/<int:message_id>', methods=['POST'])
+def toggle_like(message_id):
+    """Toggle a liked message for the current user."""
+
+    if not g.user:
+        flash("Access unauthorized.", "danger")
+        return redirect("/")
+
+    liked_message = Message.query.get(message_id)
+    if liked_message.user_id == g.user.id:
+        return abort(403)
+
+    user_likes = g.user.likes
+
+    if liked_message in user_likes:
+        g.user.likes = [like for like in user_likes if like != liked_message]
+    else:
+        g.user.likes.append(liked_message)
```

Comment:
A more efficient way to do this is to query for the Like instance that belong to the user and the message. If that doesn't exist, create it, else delete it.